xml/ftp_banners.xml

```diff
@@ -183,28 +186,58 @@ example.com FTP server (Version:  Mac OS X Server) ready.</example>
-    <example>ProFTPD 1.2.10 Server (Main FTP Server) [host]</example>
-    <example>ProFTPD 1.2.10 Server (ProFTPD) [host]</example>
-    <example>ProFTPD 1.2.10rc3 Server (ProFTPD Default Installation) [host]</example>
+    <example service.version="1.2.10">ProFTPD 1.2.10 Server (Main FTP Server) [host]</example>
+    <example proftpd.server.name="ProFTPD">ProFTPD 1.2.10 Server (ProFTPD) [host]</example>
+    <example host.name="host">ProFTPD 1.2.10rc3 Server (ProFTPD Default Installation) [host]</example>
     <param pos="0" name="service.family" value="ProFTPD"/>
+    <param pos="0" name="service.vendor" value="ProFTPD Project"/>
     <param pos="0" name="service.product" value="ProFTPD"/>
     <param pos="1" name="service.version"/>
     <param pos="2" name="proftpd.server.name"/>
     <param pos="3" name="host.name"/>
   </fingerprint>
   <fingerprint pattern="^ProFTPD (\d+\.[^\s]+) Server ready\.$">
     <description>ProFTPD with only version info</description>
-    <example>ProFTPD 1.3.0rc2 Server ready.</example>
+    <example service.version="1.3.0rc2">ProFTPD 1.3.0rc2 Server ready.</example>
     <param pos="0" name="service.family" value="ProFTPD"/>
+    <param pos="0" name="service.vendor" value="ProFTPD Project"/>
     <param pos="0" name="service.product" value="ProFTPD"/>
     <param pos="1" name="service.version"/>
   </fingerprint>
   <fingerprint pattern="^ProFTPD FTP Server ready\.$">
     <description>ProFTPD with no version info</description>
     <example>ProFTPD FTP Server ready.</example>
     <param pos="0" name="service.family" value="ProFTPD"/>
+    <param pos="0" name="service.vendor" value="ProFTPD Project"/>
     <param pos="0" name="service.product" value="ProFTPD"/>
   </fingerprint>
+  <fingerprint pattern="^ProFTPD Server$">
+    <description>ProFTPD with no version info, short form</description>
+    <example>ProFTPD Server</example>
+    <param pos="0" name="service.family" value="ProFTPD"/>
+    <param pos="0" name="service.vendor" value="ProFTPD Project"/>
+    <param pos="0" name="service.product" value="ProFTPD"/>
+  </fingerprint>
+  <fingerprint pattern="^(?:\d{4}\-\d\d\-\d\d \d\d:\d\d:\d\d,\d\d\d )?(\S+) proftpd\[\d+\]: error: no valid servers configured">
+    <description>ProFTPD no valid servers configured</description>
+    <example host.name="ftp.host.com">ftp.host.com proftpd[40312]: error: no valid servers configured\n</example>
+    <example host.name="hostname.com">2016-10-31 12:14:35,524 hostname.com proftpd[26992]: error: no valid servers configured\n</example>
+    <param pos="0" name="service.family" value="ProFTPD"/>
+    <param pos="0" name="service.vendor" value="ProFTPD Project"/>
+    <param pos="0" name="service.product" value="ProFTPD"/>
+    <param pos="1" name="host.name"/>
+  </fingerprint>
+  <fingerprint pattern="^ProFTPD (\d+\.[^\s]+) Server \((.*)\) \[[\h.:\]]*$">
+    <description>ProFTPD with version info - truncated</description>
```

Comment:
Based on the thousands of these in the dataset and number of times that some with the exact same truncation occurred I'm inclined to think this isn't an issue with the scans.io dataset.